src/helper/fast-getter.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Returns a wrapper function that can be added directly to a characteristic that
+ * will ensure that the plugin responds quickly to get requests.
+ * @param {function} actualGetter 
+ * @param {} log 
+ * @returns 
+ */
+module.exports = function createFastGetter(actualGetter, log) {
+    let lastValue = null;
+
+    return function(callback) {
+        const characteristic = this;
+
+        let timeoutResponded = false;
+
+        /* Respond with a cached value if the actual getter doesn't respond quickly */
+        const timeout = setTimeout(function() {
+            timeoutResponded = true;
+            if (lastValue) {
+                log.debug(`${characteristic.displayName}: timeout returning last result: ${lastValue.error} ${lastValue.result}`);
+                callback(lastValue.error, lastValue.result);
+            } else {
+                log.debug(`${characteristic.displayName}: timeout returning no result`);
+                callback(new Error('Device slow to respond'), null);
+            }
+        }, 500);
+
+        actualGetter(function(error, result) {
+            lastValue = { error, result };
+
+            if (!timeoutResponded) {
+                clearTimeout(timeout);
+                log.debug(`${characteristic.displayName}: returning actual result ${error} ${result}`);
+                callback(error, result);
+            } else if (error) {
+                log.debug(`${characteristic.displayName}: updating characteristic with error for late result: ${error}`);
+                characteristic.updateValue(error);
```

Comment:
Why call `updateValue` when an error occurred?

Reply:
@datMaffin hmm. Good question. I figured it might be useful to report that things are broken rather than just continuing to report the cached value like it's fine?

Reply:
Ok, interesting I did not know that `updateValue` could take an error value: https://github.com/homebridge/HAP-NodeJS/blob/master/src/lib/Characteristic.ts#L1330

I am still a bit skeptical though :wink:. 
I assume that you had this code running for a while: did this PR behave for you as expected?

Reply:
@datMaffin I have had it running for the last 10 days or so... I am definitely not getting the "plugin is slowing" messages any more... and is all still working. I am not sure whether I've hit that error path though.

My _presumption_ was that passing an error up would result in some sort of "failed" state in the Home app... it would of course be easy to test. I'm not by my homebridge at the moment, but will be.